File: examples/typescript/research.ts
import { TnlClient } from '@theneuralledger/sdk';

const apiKey = process.env.TNL_API_KEY;
if (!apiKey) throw new Error('TNL_API_KEY is required');

const client = new TnlClient({ apiKey });
const page = await client.searchNews({
  query: 'semiconductor export restrictions',
  publishedSince: new Date(Date.now() - 7 * 86_400_000).toISOString(),
  pageSize: 50,
  include: ['sources', 'claims'],
});

for (const story of page.data) {
  console.log({
    title: story.title,
    assets: story.impactedAssets,
    paths: story.impactPaths,
    confidence: story.truthPosterior,
  });
}

File: examples/python/research.py
import os
from datetime import UTC, datetime, timedelta

from tnl_intelligence import TnlClient


with TnlClient(os.environ["TNL_API_KEY"]) as client:
    page = client.search_news(
        "semiconductor export restrictions",
        published_since=(datetime.now(UTC) - timedelta(days=7)).isoformat(),
        page_size=50,
        include=["sources", "claims"],
    )
    for story in page.data:
        print(story.title, story.impacted_assets, story.impact_paths, story.truth_posterior)
